### PR TITLE
storage: Increase the timeout of storage REST requests

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
This commit increases storage REST requests to 5 minutes, this includes
the opening TCP connection, and sending/receiving data. This will reduce
clients receiving errors when the server is under high load.

## Motivation and Context
Reduce the likewise of S3 clients having errors when the server is a bit of high load.

## Regression
No

## How Has This Been Tested?
1. Start a Minio cluster of 12 nodes
2. Run this following script

```
package main

import (
	"bytes"
	"fmt"
	"log"
	"math"
	"sync"
	"time"

	"github.com/minio/minio-go"
	"github.com/minio/minio-go/pkg/encrypt"
)

func main() {

	const parallelWorkers = 32
	const hostName = "minio"

	var wg sync.WaitGroup
	var totalTransfer int64

	objectContent := bytes.Repeat(bytes.Repeat([]byte("A"), int(math.Pow(2, 19))), 1)
	objectSize := int64(len(objectContent))


	for i := 1; i <= parallelWorkers; i++ {
		wg.Add(1)
		go func(i int) {
			defer wg.Done()

			hostAddr := fmt.Sprintf("%s:%d", hostName, 9001+i%12)

			s3Client, err := minio.New(hostAddr, "minio", "minio123", true)
			if err != nil {
				log.Fatalln(err)
			}

			bucketName := "testbucket"
			objectName := fmt.Sprintf("my-object-%d", i)

			password := "correct horse battery staple"
			encryption := encrypt.DefaultPBKDF([]byte(password),
				[]byte(bucketName+objectName))

			for {
				startTime := time.Now()
				_, err := s3Client.PutObject(bucketName,
					objectName,
					bytes.NewReader(objectContent),
					objectSize,
					minio.PutObjectOptions{
						ServerSideEncryption: encryption,
					},
				)
				if err != nil {
					log.Fatalln(err)
				} else {
					totalTransfer += objectSize
				}
				elapsed := time.Since(startTime)
				if elapsed > 3*time.Second {
					log.Println(elapsed)
				}
			}
		}(i)
	}

	wg.Wait()
}
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Review by harshavardhana — CHANGES_REQUESTED** [review]
We should also do this change for rpc/* @vadmeste

**Comment by vadmeste** [comment]
> We should also do this change for rpc/* @vadmeste

Done

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

File: cmd/rpc.go [hunk 1]
```diff
@@ -217,7 +220,7 @@ func (client *RPCClient) Call(serviceMethod string, args interface {
 		}
 
 		if isNetError(err) {
-			client.setRetryTicker(time.NewTicker(xrpc.DefaultRPCTimeout))
+			client.setRetryTicker(time.NewTicker(defaultRPCTimeout))
```
Comment (harshavardhana) [inline cmd/rpc.go]: Retry perhaps should be faster but not timeout.

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Only one comment rest looks okay.

**Comment by krishnasrinivas** [comment]
@vadmeste for dist-xl minio{1...4} setup, if S3 client is talking to minio1 and minio4's network is brought down, will it cause S3 client to hang for 5 mins?

File: cmd/rpc.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/rpc.go]: PTAL @vadmeste

**Review by harshavardhana — APPROVED** [review]

**Review by krishnasrinivas — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Pull request merged**